homeassistant/components/tado/sensor.py

```diff
@@ -290,13 +301,16 @@ def _async_update_callback(self):
     @callback
     def _async_update_device_data(self):
         """Handle update callbacks."""
-        try:
-            data = self._tado.data["device"][self.device_id]
-        except KeyError:
-            return
-
-        if self.device_variable == "tado bridge status":
-            self._state = data.get("connectionState", {}).get("value", False)
+        for device in self._tado.devices:
+            if device["serialNo"] == self.device_id:
+                self._device_info = device
+
+        if self.device_variable == "battery state":
+            self._state = self._device_info["batteryState"]
+        elif self.device_variable == "connection state":
+            self._state = self._device_info.get("connectionState", {}).get(
```

Comment:
Is this a binary state? Then it should be a binary sensor entity?

Reply:
Yes, the Tado API only returns True or False, along with a timestamp of the last status report.

Reply:
@MartinHjelmare BTW, this is already present in the current code as a non-binary sensor. However, it's interesting that it wasn't a problem until I created my PR...

Reply:
There's loads of legacy code that doesn't follow our current standards.